### ISSUE TITLE
bug in running tutorial

### ISSUE THREAD

**Issue by ajian-he** [tbepler/topaz#42]
I install v0.2.3 when I run the tutorial, as well as my own data, it always come to the error like below:
UserWarning: Couldn't retrieve source code for container of type LinearClassifier. It won't be checked for correctness upon loading.
UserWarning: Couldn't retrieve source code for container of type ResNet8. It won't be checked for correctness upon loading.
UserWarning: Couldn't retrieve source code for container of type Sequential. It won't be checked for correctness upon loading.
UserWarning: Couldn't retrieve source code for container of type BasicConv2d. It won't be checked for correctness upon loading.
UserWarning: Couldn't retrieve source code for container of type Conv2d. It won't be checked for correctness upon loading.
UserWarning: Couldn't retrieve source code for container of type ReLU. It won't be checked for correctness upon loading.
UserWarning: Couldn't retrieve source code for container of type ResidA. It won't be checked for correctness upon loading.
and the program stuck!
please help! Thanks!

**Comment by tbepler** [comment]
This sounds like a problem with your install path. How did you install topaz? What system are you on?

**Comment by ajian-he** [comment]
Thanks. I follow the instruction, first I install Anaconda, then create a conda environment for Topaz as suggested "conda create -n topaz python=3.6...", finally I install topaz via "conda install topaz -c tbepler -c pytorch" . The system is CentOS 7.6. Really need help, thanks!

**Comment by tbepler** [comment]
What version of pytorch do you have installed? You can find this if you activate your conda topaz environment and run conda list:
```
source activate topaz
conda list
```

**Comment by ajian-he** [comment]
here is the outcome, thanks!

# packages in environment at /share/XXX/anaconda3/envs/topaz:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main
ca-certificates           2019.11.27                    0
certifi                   2019.11.28               py37_0
libedit                   3.1.20181209         hc058e9b_0
libffi                    3.2.1                hd88cf55_4
libgcc-ng                 9.1.0                hdf63c60_0
libstdcxx-ng              9.1.0                hdf63c60_0
ncurses                   6.1                  he6710b0_1
openssl                   1.1.1d               h7b6447c_3
pip                       19.3.1                   py37_0
python                    3.7.5                h0371630_0
readline                  7.0                  h7b6447c_5
setuptools                42.0.2                   py37_0
sqlite                    3.30.1               h7b6447c_0
tk                        8.6.8                hbc83047_0
wheel                     0.33.6                   py37_0
xz                        5.2.4                h14c3975_4
zlib                      1.2.11               h7b6447c_3

**Comment by tbepler** [comment]
It looks like topaz isn't installed into your topaz conda environment. Try running 'source activate topaz' and then 'conda install ...' again. That should fix the problem.

**Comment by ajian-he** [comment]
I tried to de novo install all conda and topaz, follow the suggestion of first activate topaz and then conda install..., the issue still exist, anyway, I can continue all tutorial, but the outcome is slightly different from the demo tutorial

**Comment by tbepler** [comment]
There is some randomness in the results due to how the neural network parameters are optimized. Are you getting the same warning, but the everything otherwise runs normally?

**Comment by ajian-he** [comment]
the training results are different from the tutorial:
epoch	iter	split	loss	ge_penalty	precision	tpr	fpr	auprc
1000	1	1001	test	0.015258	-	0.040877	0.258628	0.009914	0.05945546523116192
2001	2	2001	test	0.015993	-	0.044977	0.294445	0.010214	0.061771123509967794
3002	3	3001	test	0.020187	-	0.042730	0.374459	0.013705	0.06958290330218277
4003	4	4001	test	0.020249	-	0.045742	0.388267	0.013233	0.0675878506697664
5004	5	5001	test	0.021257	-	0.047583	0.420393	0.013747	0.0704535301988471
6005	6	6001	test	0.022189	-	0.040423	0.400547	0.015534	0.07013125083592019
7006	7	7001	test	0.024289	-	0.050577	0.467966	0.014352	0.07362886436482818
8007	8	8001	test	0.019194	-	0.055014	0.402074	0.011283	0.07455601152215174
9008	9	9001	test	0.018753	-	0.049260	0.378237	0.011926	0.07213143960590812
10009	10	10001	test	0.017719	-	0.058193	0.401393	0.010613	0.0823831178371298

**Comment by ajian-he** [comment]
and the picked particles is much less than tutorial

**Comment by tbepler** [comment]
As I said, there is randomness from the optimizer. These results look normal. Just lower the score threshold to get more particles. In the tutorial, this threshold is set in the "topaz particle_stack" and micrograph visualization steps.

**Comment by ajian-he** [comment]
thanks, I am running my own data now, let's see how it lood

**Comment by ajian-he** [comment]
tbepler, I found that topaz indeed found many previous hided or unfound particles, however, I have issues either use the coordinates from topaz to extract particles in relion, or use the topaz extracted particles to do further analysis (e.g. class 2d, 3d) in relion as they lack ctf or defocus values. I try to use the relion micrographs_ctf.star as metadata to combine the data, but failed. Any suggestion, please. Thank you.

**Comment by tbepler** [comment]
What do you mean by it failed? Did you get an error from "topaz particle_stack" when you tried to include your metadata?

I believe you can load the particles directly into relion (using the .star file from extract) and add CTF information there. @alexjnoble can you chime in on adding CTF information in relion?

**Comment by ajian-he** [comment]
when I use import job in relion to import the topaz particle256.star, the import job seems ok, but when I later use the import to do 2D classification, it shows error like this:
ERROR: 
Cannot read file particles256.mrc It does not exist
[localhost.localdomain:101565] 3 more processes have sent help message help-mpi-api.txt / mpi-abort
[localhost.localdomain:101565] Set MCA parameter "orte_base_help_aggregate" to 0 to see all help / error messages

when I tried to directly load particle256.star into relion 2D classification, it also shows :
ERROR: 
Cannot read file particles256.mrc It does not exist

Could you give me a step-by-step guide how to load the topaz extract to relion, either the particle.star file way, or the coordinate file way? Thanks a lot!

**Comment by ajian-he** [comment]
Figured out, use re-extract, thanks!

**Comment by tbepler** [comment]
Great, I'm glad to hear it's working. I'm going to close this issue since it sounds like everything is resolved.
